gluten-core/src/main/scala/org/apache/gluten/GlutenPlugin.scala

```diff
@@ -135,6 +135,37 @@ private[gluten] class GlutenDriverPlugin extends DriverPlugin with Logging {
     }
   }
 
+  private def checkOffHeapSettings(conf: SparkConf): Unit = {
+    if (
+      conf.getBoolean(
+        DYNAMIC_OFFHEAP_SIZING_ENABLED.key,
+        DYNAMIC_OFFHEAP_SIZING_ENABLED.defaultValue.get)
+    ) {
+      // When dynamic off-heap sizing is enabled, off-heap mode is not strictly required to be
+      // enabled. Skip the check.
+      return
+    }
+
+    if (
+      conf.getBoolean(COLUMNAR_MEMORY_UNTRACKED.key, COLUMNAR_MEMORY_UNTRACKED.defaultValue.get)
```

Comment:
looking at the logic, if `DYNAMIC_OFFHEAP_SIZING_ENABLED=false`, and `COLUMNAR_MEMORY_UNTRACKED=true` then it will also skip the check of off-heap settings, is this inteded?

Reply:
Yes, as mentioned in the other comment, it's allowed for user to set

```
spark.memory.offHeap.enabled=false
spark.gluten.memory.untracked=true
```

at the same time.